content/articles/a-gentle-introduction-to-the-python-binance-api/index.md

```diff
@@ -0,0 +1,335 @@
+
+```python
+numeric_columns = ['Open', 'High', 'Low', 'Close', 'Volume', 'Quote Asset Volume', 'TB Base Volume', 'TB Quote Volume']
+hist_df[numeric_columns] = hist_df[numeric_columns].apply(pd.to_numeric, axis=1)
+```
+
+We have created a variable called `numeric_columns`. This variable stores the list of the data we want to convert into numeric. We then use the `apply()` method to perform this converstion. 
+
+We can take a look at the results by issuing the command, `hist_df.head()`:
+
+![Numerical columns](/engineering-education/a-gentle-introduction-to-the-python-binance-api/numerical-columns.png)
+
+After transforming the data types, we can view the new data types using the command, `hist_df.dtypes`:
+
+```bash
+Open Time             datetime64[ns]
+Open                         float64
+High                         float64
+Low                          float64
+Close                        float64
+Volume                       float64
+Close Time            datetime64[ns]
+Quote Asset Volume           float64
+Number of Trades               int64
+TB Base Volume               float64
+TB Quote Volume              float64
+Ignore                        object
+dtype: object
+```
+We've successfully preprocessed the historical data. Additionally, you can do other types of analysis on the data. Play around with `hist_df.describe()` and `hist_df.info` to see what you can find.
+
+### Visualization using matplotlib finance library
+In this step, we will explore and visualize data using [mplfinance](https://github.com/matplotlib/mplfinance). It is a financial market data visualization tool built on top of matplotlib.
+
+We begin by importing the library into our notebook.
+
+```python
+import mplfinance as mpf
+```
+We will visualize the `Close Time` data and limit the data to only the last 100 rows of data. In the plot, we've also included the candlestick type of plotting, volume data, the title for our plot, and finally, the 10, 20, and 30 moving day averages.
+
+```python
+mpf.plot(hist_df.set_index('Close Time').tail(100), 
+        type='candle', style='charles', 
+        volume=True, 
+        title='BTCUSDT Last 100 Days', 
+        mav=(10,20,30))
+```
+Output:
+
+![Visualization](/engineering-education/a-gentle-introduction-to-the-python-binance-api/plot.png)
+
+As we've seen, the matplotlib finance library allows us to visualize data easily.
+
+Please find the code for this tutorial [here](https://colab.research.google.com/drive/1MP-Kk4gG-fnbCK40MxENCdIKr0KRho9f?usp=sharing).
+
+### Summary
+In a nutshell, this tutorial has shown you what's possible with the python-binance API. We've shown you how you can perform an exploratory data analysis on Binance data. To avoid making the tutorial so long, we've only limited it to the EDA bit. In our follow-up tutorial, we'll build upon this tutorial and show you how to apply machine learning and data science to trading and finance.
```

Comment:
The third sentence in this paragraph seems off. Consider restructuring it or removing it entirely.